src/service/trello-api-service.ts

```diff
@@ -0,0 +1,204 @@
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        text: comment
+      }),
+    });
+
+    if (!response.ok) {
+      throw new Error(
+        `Failed to create comment: ${response.status} - ${response.statusText}`
+      );
+    }
+    return await response.json();
+  }
+
+  /**
+   * Deletes a card on the specified Trello list
+   *
+   * @param cardId card ID
+   * @return deletion result
+   */
+  public async deleteCard(cardId: string): Promise<any> {
+    const response = await fetch(`${this.baseUrl}/cards/${cardId}?key=${this.apiKey}&token=${this.apiToken}`, {
+      method: "DELETE",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        id: cardId
+      }),
+    });
+
+    if (!response.ok) {
+      throw new Error(
+        `Failed to delete card: ${response.status} - ${response.statusText}`
+      );
+    }
+    return await response.json();
+  }
+
+  /**
+   * Lists Trello lists
+   *
+   * @returns lists
+   */
+  public async getListsOnBoard(): Promise<TrelloList[]> {
+    const response = await fetch(`${this.baseUrl}/boards/${this.boardId}/lists?key=${this.apiKey}&token=${this.apiToken}`, {
+      method: "GET",
+    });
+  
+    if (!response.ok) throw new Error(`Failed to fetch lists: ${response.status} - ${response.statusText}`);
+    const listsData = await response.json();
+    const lists: TrelloList[] = listsData.map((list: any) => ({
+      id: list.id,
+      name: list.name,
+      idBoard: list.idBoard,
+    }));
+    console.log("lists:", lists)
+    return lists;
+  }
+
+  /**
+   * Lists specified Trello list
+   *
+   * @param name  list name
+   * @returns list content
+   */
+  public getListIdByName = async (name: string) => {
+    const lists = await this.getListsOnBoard();
+    const listFound = lists.find(list => list.name == name);
+    return listFound.id;
+  }
+
+  /**
+   * Lists Trello cards on a list
+   *
+   * @param listId  Trello ID list
+   * @returns cards
+   */
+    public async getCardsOnList(): Promise<TrelloCards[]> {
+      const listId = await this.getListIdByName("Memo");
+      const url = `${this.baseUrl}/lists/${listId}/cards?key=${this.apiKey}&token=${this.apiToken}`;
+  
+      const response = await fetch(url, {
+        method: "GET",
+        headers: { "Content-Type": "application/json" },
+      });
+      const data = await response.json();
+      
+      return data.map((card: any) => ({
+        cardId: card.shortLink,
+        title: card.name,
+        description: card.desc,
+        assignedPersons: card.idMembers || [],
+      }));
+    }
+
+  /**
+   * Fetches email addresses of all users related to a specific board
+   *
+   * @returns emails
+   */
+  public async getBoardMembers(): Promise<TrelloMembers[]> {
```

Comment:
Why do we need this?

Reply:
I expected that it might be usefull on UI to have assigned persons' email, name and id related to cards.